src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ConfigurationGeneral.xaml

```diff
@@ -52,6 +52,9 @@
   <StringProperty Name="Configuration"
                   Visible="false" />
 
+  <StringProperty Name="WindowsFormsDesignerMode"
```

Comment:
Why not define this as EnumProperty with the values here, and generated enum, rather than handle the mapping yourself and creating the enum yourself?

Reply:
Does it actually generate a enum? They were still strings.

Reply:
It should generate an enum (called WindowsFormsDesignerModeValues I guess) in the .cs file in the obj directory. From memory the values still get read as strings, but the cps property providers take care of checking for valid values, so consuming code can just use `string.Equals` for a cheaper comparison. It's possible that last statement is wrong however, and that's just the pattern of how CPS does it.

Reply:
i'll look into it, it actually looks a lot more complicated than just reading a string value and parsing.